polyfill/index.d.ts

```diff
@@ -0,0 +1,385 @@
+    minus(temporalDurationLike: DurationLike, options?: ArithmeticOptions): Temporal.Date;
+    difference(other: Temporal.Date, options?: DifferenceOptions<'years' | 'months' | 'days'>): Temporal.Duration;
+    withTime(temporalTime: Temporal.Time): Temporal.DateTime;
+    getYearMonth(): Temporal.YearMonth;
+    getMonthDay(): Temporal.MonthDay;
+    getFields(): Required<DateLike>;
+    toLocaleString(locales?: string | string[], options?: Intl.DateTimeFormatOptions): string;
+    toJSON(): string;
+    toString(): string;
+  }
+
+  export type DateTimeLike = {
+    year?: number;
+    month?: number;
+    day?: number;
+    hour?: number;
+    minute?: number;
+    second?: number;
+    millisecond?: number;
+    microsecond?: number;
+    nanosecond?: number;
+  };
+
+  export class DateTime extends Required<DateTimeLike> {
+    static from(item: Temporal.DateTime | string | object, options?: AssignmentOptions): Temporal.DateTime;
+    static compare(one: Temporal.DateTime, two: Temporal.DateTime): ComparisonResult;
+    constructor(
+      year: number,
+      month: number,
+      day: number,
+      hour?: number,
+      minute?: number,
+      second?: number,
+      millisecond?: number,
+      microsecond?: number,
+      nanosecond?: number
+    );
+    readonly year: number;
+    readonly month: number;
+    readonly day: number;
+    readonly hour: number;
+    readonly minute: number;
+    readonly second: number;
+    readonly millisecond: number;
+    readonly microsecond: number;
+    readonly nanosecond: number;
+    readonly dayOfWeek: number;
+    readonly dayOfYear: number;
+    readonly weekOfYear: number;
+    readonly daysInYear: number;
+    readonly daysInMonth: number;
+    readonly isLeapYear: boolean;
+    with(temporalDateTimeLike: DateTimeLike, options?: AssignmentOptions): Temporal.DateTime;
+    plus(temporalDurationLike: DurationLike, options?: ArithmeticOptions): Temporal.DateTime;
+    minus(temporalDurationLike: DurationLike, options?: ArithmeticOptions): Temporal.DateTime;
+    difference(
+      other: Temporal.DateTime,
+      options?: DifferenceOptions<'years' | 'months' | 'days' | 'hours' | 'minutes' | 'seconds'>
+    ): Temporal.Duration;
+    // Should Absolute and DateTime have different names for `inTimeZone`? See #574.
+    inTimeZone(temporalTimeZoneLike: TimeZoneLike, options?: ToAbsoluteOptions): Temporal.Absolute;
+    getDate(): Temporal.Date;
+    getYearMonth(): Temporal.YearMonth;
+    getMonthDay(): Temporal.MonthDay;
+    getTime(): Temporal.Time;
+    getFields(): Required<DateTimeLike>;
+    toLocaleString(locales?: string | string[], options?: Intl.DateTimeFormatOptions): string;
+    toJSON(): string;
+    toString(): string;
+  }
+
+  export type MonthDayLike = {
+    month?: number;
+    day?: number;
+  };
+
+  export class MonthDay extends Required<MonthDayLike> {
+    static from(item: Temporal.MonthDay | string | object, options?: AssignmentOptions): Temporal.MonthDay;
```

Comment:
```suggestion
    static from(item: MonthDayLike | string | object, options?: AssignmentOptions): Temporal.MonthDay;
```

Reply:
Yep. Per earlier discussion, I'll make this `Temporal.MonthDay | MonthDayLike`

Reply:
Circling back, here's where it ended up:
* Parameters of methods like `plus` that accept a plain-JS property bag will get this signature: `Xxx | XxxLike`. This is redundant, because Temporal Types (except TimeZone and Absolute) all implement their Like interfaces. But I think this redundancy will help users who look at the types in IDE tooltip help, because it makes it easier to realize that they don't have to call `getFields()` to call the method but instead can just pass the class instance.
* Exception: `with` methods accept a Temporal class instance, but that seems like an anti-pattern (users who want to clone an object should use `from`), so I made the signature just `XxxLike` to emphasize that the property bag is the main use case for that method. Passing a class instance won't trigger a TS warning (because every instance type implements its Like interface) but may be useful to guide developers towards best practices.

Holler if you disagree.